clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -4675,6 +4675,7 @@ bool SYCLIntegrationFooter::emit(raw_ostream &OS) {
     VD = VD->getCanonicalDecl();
     if (VD->isInAnonymousNamespace()) {
       std::string TopShim = EmitSpecIdShims(OS, ShimCounter, VD);
+      OS << "__SYCL_INLINE_NAMESPACE(cl) {\n";
       OS << "namespace sycl {\n";
       OS << "namespace detail {\n";
```

Comment:
Are we to the point where we should hoist these three from the `if` and `else` branches?

Reply:
Ah... we can't.  The EmitSpecIdShims on line 4677 emits a bunch of stuff.  I wonder if I could add an isInAnonymousNamespace check to that emit, then simplify this function a bit...

Reply:
You know, I think that's caught me twice now....

I'm fine leaving the code as-is, it was more of a drive-by (stupid) comment.

Reply:
I refactored to be more clear and remove some of the duplication.  The duplication already caught me 1x today, plus now you 2-3 times :)  I think it is worth doing.